cmd/list.go

```diff
@@ -0,0 +1,87 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"golang.org/x/mod/modfile"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+func ListCmd() *cobra.Command {
+	var listCmd = &cobra.Command{
+		Use:   "list",
+		Short: "List the docker images",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdFlags := cmd.Flags()
+
+			configFile, _ := cmdFlags.GetString(flagFile)
+			if configFile == "" {
+				// try to load a local chains.yaml, but do not panic for any error, will fall back to embedded chains.
+				cwd, err := os.Getwd()
+				if err == nil {
+					chainsYamlSearchPath := filepath.Join(cwd, "chains.yaml")
+					if err := loadChainsYaml(chainsYamlSearchPath); err != nil {
+						fmt.Printf("No config found at %s, using embedded chains. pass -f to configure chains.yaml path.\n", chainsYamlSearchPath)
+					} else {
+						fmt.Printf("Loaded chains from %s\n", chainsYamlSearchPath)
+					}
+				}
+			} else {
+				// if flag is explicitly provided, panic on error since intent was to override embedded chains.
+				if err := loadChainsYaml(configFile); err != nil {
+					panic(err)
+				}
+			}
```

Comment:
In Go, it's idiomatic to:
* Limit use of `else` statements. Instead, return early. 
* Don't nest the "happy path" via `err == nil`

This limits nesting and makes it more linear and easier to read. 

Note: This suggestion hasn't been tested and may not compile. 

```suggestion
# Refactor this into a private function. 
	if configFile == "" {
		// try to load a local chains.yaml, but do not panic for any error, will fall back to embedded chains.
		cwd, err := os.Getwd()
		if err != nil {
			panic(err)
		}
		chainsYamlSearchPath := filepath.Join(cwd, "chains.yaml")
		err = loadChainsYaml(chainsYamlSearchPath)
		if err != nil {
			fmt.Printf("No config found at %s, using embedded chains. pass -f to configure chains.yaml path.\n", chainsYamlSearchPath)
			return
		}
		fmt.Printf("Loaded chains from %s\n", chainsYamlSearchPath)
		return
	}
	// if flag is explicitly provided, panic on error since intent was to override embedded chains.
	if err := loadChainsYaml(configFile); err != nil {
		panic(err)
	}
```

Reply:
This was copy/paste from `buildCmd()` didn't put much thought into it